File: 1038-binary-search-tree-to-greater-sum-tree/1038-binary-search-tree-to-greater-sum-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def bstToGst(self, root: TreeNode) -> TreeNode:
        
        def change(node):
            nonlocal tot
            if(not node):
                return 
            
            change(node.right)
            tot += node.val
            node.val = tot
            change(node.left)
        
        tot = 0
        change(root)
        return root
            

File: 1038-binary-search-tree-to-greater-sum-tree/1038-binary-search-tree-to-greater-sum-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func bstToGst(root *TreeNode) *TreeNode {
    
    var incrementNode func(node *TreeNode, val int) int;
    
    incrementNode = func(node *TreeNode, val int) int {
        if node == nil {
            return val
        }
    
        var rightVal = incrementNode(node.Right, val)
        node.Val += rightVal
        return incrementNode(node.Left, node.Val)
    }
    
    incrementNode(root, 0)
    return root
}